client/blocks/inline-help/admin-sections.js

```diff
@@ -0,0 +1,430 @@
+		icon: 'credit-card',
+	},
+	{
+		title: translate( 'Download the WordPress.com app for my device' ),
+		description: translate( 'Get WordPress apps for all your screens.' ),
+		link: '/me/get-apps',
+		synonyms: [ 'android', 'iphone', 'mobile', 'desktop', 'phone' ],
+		icon: 'my-sites',
+	},
+	{
+		title: translate( 'View my drafted posts' ),
+		link: `/posts/drafts/${ siteSlug }`,
+		synonyms: [ 'posts', 'draft' ],
+		icon: 'my-sites',
+	},
+	{
+		title: translate( 'Manage my blog posts' ),
+		link: '/posts/${ siteSlug }',
+		synonyms: [ 'lists', 'posts' ],
+		icon: 'my-sites',
+	},
+	{
+		title: translate( 'View my drafted pages' ),
+		link: `/pages/drafts/${ siteSlug }`,
+		synonyms: [ 'pages', 'draft' ],
+		icon: 'my-sites',
+	},
+	{
+		title: translate( 'Manage my pages' ),
+		link: `/pages/${ siteSlug }`,
+		synonyms: [ 'lists', 'pages' ],
+		icon: 'my-sites',
+	},
+	{
+		title: translate( 'I cannot find my site on Google' ),
+		link: `/marketing/traffic/${ siteSlug }`,
+		synonyms: [ 'google', 'traffic' ],
+		icon: 'speaker',
+	},
+	{
+		title: translate( 'Verify my site with Google' ),
+		link: `/marketing/traffic/${ siteSlug }`,
+		synonyms: [ 'google', 'traffic' ],
+		icon: 'cog',
+	},
+	{
+		title: translate( 'Create a new site' ),
+		link: '/jetpack/new?ref=calypso-selector',
+		synonyms: [ 'site' ],
+		icon: 'cog',
+	},
+	{
+		title: translate( 'View contact form messages' ),
+		link: `https://${ siteSlug }/wp-admin/edit.php?post_type=feedback&calypsoify=1`,
+		synonyms: [ 'contact', 'form' ],
+		icon: 'cog',
+	},
+	{
+		title: translate( 'Portfolio projects (for those who have them active)' ),
+		link: `/types/jetpack-portfolio/${ siteSlug }`,
+		synonyms: [ 'portfolio' ],
+		icon: 'cog',
+	},
+] );
+
+/**
+ * Returns a filtered site admin collection.
+ *
+ * @param   {string} searchTerm The search term
+ * @param   {Array}  collection A collection of site admin objects
+ * @param   {number} limit      The maximum number of results to show
+ * @returns {Array}             A filtered (or empty) array
+ */
+export function filterListBySearchTerm( searchTerm = '', collection = [], limit = 4 ) {
+	const searchTermWords = words( searchTerm ).map( ( word ) => word.toLowerCase() );
+
+	if ( searchTermWords.length < 1 ) {
+		return [];
+	}
+
+	const searchRegex = new RegExp(
+		// Join a series of look aheads
+		// matching full and partial works
+		// Example: "Add a dom" => /(?=.*\badd\b)(?=.*\ba\b)(?=.*\bdom).+/gi
+		searchTermWords
+			.map( ( word, i ) =>
+				// if it's the last word, don't look for a end word boundary
+				// otherwise
+				i + 1 === searchTermWords.length ? `(?=.*\\b${ word })` : `(?=.*\\b${ word }\\b)`
+			)
+			.join( '' ) + '.+',
+		'gi'
+	);
+
+	return collection
+		.filter( ( item ) => {
+			if ( searchRegex.test( item.title ) ) {
+				return true;
+			}
+			// Until we get the synonyms translated, just check when the language is `'en'`
+			return 'en' === getLocaleSlug()
```

Comment:
Minor: using `getLocaleSlug` here could ultimately make this function more difficult to test. The dependency could be injected to make the function purer.

Reply:
As discussed this can be addressed later so we can get this PR in as it's starting to block other work on this feature.